corehq/apps/users/models.py

```diff
@@ -1515,6 +1515,11 @@ def _get_viewable_report_slugs(self, domain):
     def can_view_some_reports(self, domain):
         return self.can_view_reports(domain) or bool(self.get_viewable_reports(domain))
 
+    def can_view_some_data_registry_contents(self, domain):
+        dm = self.get_domain_membership(domain, allow_enterprise=True)
+        data_registry_contents_list = dm.permissions.view_data_registry_contents_list if dm else []
+        return self.can_view_data_registry_contents() or (domain in data_registry_contents_list)
```

Comment:
just check if there are any item in the list:
```suggestion
        return self.can_view_data_registry_contents() or bool(data_registry_contents_list)
```

Note: it is a list of registry slugs not domains.

While I think it's OK to have this here I also want to point out that there is already a permission helper class for DR: [RegistryPermissionCheck](https://github.com/dimagi/commcare-hq/blob/9ee3e42e1f1e0a69e2dc64eac49c3d8235b2374f/corehq/apps/registry/utils.py#L32). It has the same logic as this but for managing DRs.

I think the only advantage to putting it all in `RegistryPermissionCheck` is that it's encapsulated and doesn't bloat the user model. I do realize that it sets up a different pattern for doing permission checks but in this case I think it's OK and allows us to keep the DR code in one place (as much as possible).

Reply:
ah thanks I agree with keeping it all in `RegistryPermissionCheck`-- [ba09a3a](https://github.com/dimagi/commcare-hq/pull/31067/commits/ba09a3acd25a19c1b2317c3a917b5097a88cdcb9)